src/lib.rs

```diff
@@ -46,34 +46,29 @@ pub struct PrePush {
     pub remote_location: String,
 }
 
-
-pub fn prepare_commit_msg<F: Fs, G: Git>(opt: PrepareCommitMsg, config: Config) -> Result<(), Box<dyn Error>> {
+pub fn prepare_commit_msg<F: Fs, G: Git>(opt: PrepareCommitMsg, config: Config) -> Result<PolicyResult, Box<dyn Error>> {
     if opt.commit_source.is_none() {
-        if let Some(_) = config.prepend_branch_name {
-            prepend_branch_name::<F, G>(opt.commit_file)?;
-        }
-
-        Ok(())
+        compose(vec![
```

Comment:
Maybe call this collect?

Reply:
This triggered the idea to implement FromIterator for PolicyResult